pennylane/qchem/openfermion_obs.py

```diff
@@ -917,6 +892,124 @@ def molecular_hamiltonian(
     + (0.176276408043196) [Z2 Z3]
     """
 
+    if len(args) != 0:
+        return _molecular_hamiltonian_dispatch(*args, **kwargs)
+
+    if "symbols" in kwargs:
+        symbols = kwargs["symbols"]
+        kwargs.pop("symbols")
+        return _molecular_hamiltonian_dispatch(symbols, **kwargs)
+
+    if "molecule" in kwargs:
+        molecule = kwargs["molecule"]
+        kwargs.pop("molecule")
+        return _molecular_hamiltonian_dispatch(molecule, **kwargs)
+
+    raise NotImplementedError("Unsupported type")
+
+
+@singledispatch
+def _molecular_hamiltonian_dispatch(*args, **kwargs):
+    r"""Generate the qubit Hamiltonian of a molecule."""
+    raise NotImplementedError("Unsupported type")
+
+
+@_molecular_hamiltonian_dispatch.register(Molecule)
+def _(
+    molecule,
+    method="dhf",
+    active_electrons=None,
+    active_orbitals=None,
+    mapping="jordan_wigner",
+    outpath=".",
+    wires=None,
+    args=None,
+    load_data=False,
+    convert_tol=1e012,
+):
+    return _molecular_hamiltonian(
+        molecule.symbols,
+        molecule.coordinates,
+        molecule.name,
+        molecule.charge,
+        molecule.mult,
+        molecule.basis_name,
+        method,
+        active_electrons,
+        active_orbitals,
+        mapping,
+        outpath,
+        wires,
+        molecule.alpha,
+        molecule.coeff,
+        args,
+        load_data,
+        convert_tol,
+    )
+
+
+@_molecular_hamiltonian_dispatch.register(list)
+def _(
+    symbols,
+    coordinates,
+    name="molecule",
+    charge=0,
+    mult=1,
+    basis="sto-3g",
+    method="dhf",
+    active_electrons=None,
+    active_orbitals=None,
+    mapping="jordan_wigner",
+    outpath=".",
+    wires=None,
+    alpha=None,
+    coeff=None,
+    args=None,
+    load_data=False,
+    convert_tol=1e012,
+):
+    return _molecular_hamiltonian(
+        symbols,
+        coordinates=coordinates,
+        name=name,
+        charge=charge,
+        mult=mult,
+        basis=basis,
+        method=method,
+        active_electrons=active_electrons,
+        active_orbitals=active_orbitals,
+        mapping=mapping,
+        outpath=outpath,
+        wires=wires,
+        alpha=alpha,
+        coeff=coeff,
+        args=args,
+        load_data=load_data,
+        convert_tol=convert_tol,
+    )
+
+
+def _molecular_hamiltonian(
+    symbols=None,
+    coordinates=None,
```

Comment:
This is because we didn't want to change a lot in the old function since we want to keep old UI as it is. This is a part of tests from old function and will be deprecated eventually.